### ISSUE TITLE
[calcite-input-date-picker] add status prop

### ISSUE THREAD

**Issue by joeyHarig** [Esri/calcite-design-system#5208]
### Description

Add a `status` prop to the `calcite-input-date-picker` component that would behave the same as the `status` prop on `calcite-input`.

![Screen Shot 2022-08-23 at 2 08 27 PM](https://user-images.githubusercontent.com/55154968/186255952-3c1076d7-2174-44c7-8da9-13ea24367e1d.png)

### Acceptance Criteria

same as the `status` prop on `calcite-input`

### Which Component

calcite-input-date-picker

**Comment by jar12039** [comment]
The Business Analyst team would also like to see this implemented. The `input-time-picker` could benefit from having `status` added as well. All other input components have this property.

**Comment by jcfranco** [comment]
Linked to validation epic. cc @geospatialem

**Comment by geospatialem** [comment]
See the [Figma file](https://www.figma.com/file/5QDPzXr21qQcPTkXR1Ouga/Form-Validation-%5BEpic-4598%5D?type=design&node-id=956-94284&mode=design&t=tPqHJcVdxmJmZ6L2-0) for implementation:

![image](https://github.com/Esri/calcite-design-system/assets/5023024/2dd20147-2f70-4deb-abc1-7980ed140c5a)

**Comment by jcfranco** [comment]
@benelan Do you have bandwidth to pick this up this week?

**Comment by geospatialem** [comment]
This issue is blocked by the efforts of https://github.com/Esri/calcite-design-system/issues/8057 and https://github.com/Esri/calcite-design-system/issues/8000, which are targeted for late 2023.

**Comment by geospatialem** [comment]
Closing in favor of the efforts completed in https://github.com/Esri/calcite-design-system/issues/8057, where the status property was added, along with an internal input-message for displaying validation messages.

Additional efforts will be completed in https://github.com/Esri/calcite-design-system/issues/8000 for setting the validity state of the component internally.
